docs/collect-troubleshooting.rst

```diff
@@ -21,3 +21,82 @@ This generally occurs with forms that have a lot of complex :doc:`logic <form-lo
+* If you're trying to do something with the ``description`` of all of the repeat instances, you could use a function like :func:`join` from outside the repeat.
+
+You can reproduce this error yourself using `this form definition <https://docs.google.com/spreadsheets/d/1GK8CNawKlDiAx9M4y3df_7gtMQPHshdQ9Bc0fUENzU8/edit?gid=1068911091#gid=1068911091>`__. Start by selecting ``Blue`` to skip the first part of the form. When you get to the repeat, start by adding a single repeat instance and notice that the form works as expected. Then, go back and add another repeat instances, navigate forward, and see this crash.
+
+.. tip::
+
+  Whenever you have a form with a repeat, make sure to test it with 0, 1 and multiple repeat instances. This can help you catch this kind of issue before you send it out to be used.
+
+.. _troubleshoot-repeated-field-dataset:
+
+Multiple values from a dataset used as a single value
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Consider this error:
+
+.. code-block:: xml
+
+  XPath evaluation: type mismatch
+  This field is repeated:
+
+  instance(my_list)/root/item[1]/label[1];instance(my_list)/root/item[3]/label[1]
+
+Look at the repeated fields in the error and notice ``instance(my_list)/root/item[1]`` and ``instance(my_list)/root/item[3]``. This means that there is a list with name ``my_list`` and that there was an attempt to read the ``label`` of the first and third items in that list.
+
+You can experience this error yourself by using `this form definition <https://docs.google.com/spreadsheets/d/1GK8CNawKlDiAx9M4y3df_7gtMQPHshdQ9Bc0fUENzU8/edit?gid=1068911091#gid=1068911091>`__. Start by selecting ``Blue`` and go to the next screen to see that the single blue item was correctly identified. Then go back to the first question and select ``Red``. Go to the next screen and you will see the crash above. The expression that causes the crash is ``instance('my_list')/root/item[color=${color}]/label`` because multiple items match the ``color=${color}`` expression when you select ``Red`` as the color.
+
+Here are some possible fixes depending on your needs:
+
+* Ensure uniqueness of the property to filter on. In the form above, you could make sure that each color is only used once in the ``things`` list. There's no built-in way to enforce this restriction in ODK but you could use a spreadsheet constraint to do so. If you're using an Entity List, you could use a constraint in your registration form.
+* Use a different item property to filter on. In the form above, if you do want multiple items to have the same ``color``, the form structure doesn't make sense. Maybe it should start by showing a list of items first instead and then any other :ref:`lookup expressions <referencing-values-in-datasets>` could filter by ``name`` which is required to be unique.
```

Comment:
s/the form structure doesn't make sense/you'll need to rethink your form design/. Then "Maybe the form should start..."

name, which ODK guarantees will be unique.